### PR TITLE
GG-25433 Test plan [GG-25433] Extend test coverage [GG-22696] [IGNITE…

### PR DESCRIPTION
**Pull request by ingvard**
…-10808] Discovery message queue may build up with TcpDiscoveryMetricsUpdateMessage

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/MetricsCompactionTest.java [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+        assertTrue(GridTestUtils.waitForCondition(() -> records.size() == 2, 10_000));
+        assertTrue(records.stream().anyMatch(m -> m.passedLaps(srv.localNode().id()) == 1 && m.metrics().containsKey(TEST_UUID)));
+        assertTrue(records.stream().anyMatch(m -> m.passedLaps(srv.localNode().id()) == 0 && m.metrics().containsKey(TEST_UUID)));
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testQueueDoesNotContainMoreTwoMetricMessages() throws Exception {
```
Comment (sk0x50) [inline modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/MetricsCompactionTest.java]: I would prefer the following name - testQueueDoesNotContainMore*Than*TwoMetricMessages

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/MetricsCompactionTest.java [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+                    latchs.get(1).await();
+                }
+                catch (InterruptedException ignore) {
+                }
+            }
+        });
+
+        latchs.put(1, new CountDownLatch(1)); // Blocks srv discovery for sending
+
+        Thread.sleep(1_000);
```
Comment (sk0x50) [inline modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/MetricsCompactionTest.java]: Why do we need to sleep here for 1 sec?

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/MetricsCompactionTest.java [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.spi.discovery.tcp.messages.TcpDiscoveryAbstractMessage;
+import org.apache.ignite.spi.discovery.tcp.messages.TcpDiscoveryMetricsUpdateMessage;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+/**
+ *
```
Comment (sk0x50) [inline modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/MetricsCompactionTest.java]: Please add javadoc here and below.

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/MetricsCompactionTest.java (hunk 1 above)
Comment (ingvard) [reply modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/MetricsCompactionTest.java]: Thanks. Fixed.

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/MetricsCompactionTest.java (hunk 3 above)
Comment (ingvard) [reply modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/MetricsCompactionTest.java]: Added

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/MetricsCompactionTest.java (hunk 2 above)
Comment (ingvard) [reply modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/MetricsCompactionTest.java]: I added another type of dummy message, it should allow remove it.

**Pull request closed without merging**